File: Problems/USACO/chapter1/sec1.2./ride.py
"""
ID: grgurin1
LANG: PYTHON3
TASK: ride
"""
fin = open ('ride.in', 'r')
fout = open ('ride.out', 'w')

comet = fin.readline().strip()
group = fin.readline().strip()

ans = 'STAY'
cans = 1
gans = 1
# Samo ascii
for c in comet:
    cans *= ord(c)-ord('A')+1
for g in group:
    gans *= ord(g)-ord('A')+1

if gans % 47 == cans % 47:
    ans = 'GO'

fout.write (ans + '\n')
fout.close()

File: Problems/USACO/chapter1/sec1.2./ride.cpp
/*
ID: grgurin1
TASK: ride
LANG: C++                 
*/
#include <iostream>
#include <fstream>
#include <string>

using namespace std;

string group, ride;

int translate(string name){
    int res = 1;
    for(auto n : name) res *= int(n) - int('A') + 1;
    return res;
}


int main() {
    ifstream fin("ride.in");
    ofstream fout("ride.out");

    fin >> group >> ride;

    if(translate(group)%47 == translate(ride)%47) {
        fout << "GO\n";
    } else {
        fout << "STAY\n";
    }

    return 0;
}